src/main/java/org/broadinstitute/hellbender/utils/pileup/ReadPileup.java

```diff
@@ -317,73 +321,59 @@ static void fixPairOverlappingQualities(final PileupElement firstElement,
     }
 
     /**
-     * Returns a new ReadBackedPileup where only one read from an overlapping read
+     * Returns a new ReadPileup where only one read from an overlapping read
      * pair is retained.  If the two reads in question disagree to their basecall,
      * neither read is retained.  If they agree on the base, the read with the higher
      * base quality observation is retained
      *
      * @return the newly filtered pileup
      */
-    @Override
-    public ReadBackedPileup getOverlappingFragmentFilteredPileup() {
+    public ReadPileup getOverlappingFragmentFilteredPileup() {
         return getOverlappingFragmentFilteredPileup(true, true);
     }
 
     /**
-     * Returns a new ReadBackedPileup where only one read from an overlapping read
+     * Returns a new ReadPileup where only one read from an overlapping read
      * pair is retained.  If discardDiscordant and the two reads in question disagree to their basecall,
      * neither read is retained.  Otherwise, the read with the higher
      * quality (base or mapping, depending on baseQualNotMapQual) observation is retained
      *
      * @return the newly filtered pileup
      */
-    @Override
-    public ReadBackedPileupImpl getOverlappingFragmentFilteredPileup(boolean discardDiscordant, boolean baseQualNotMapQual) {
-        if (pileupElementTracker instanceof PerSamplePileupElementTracker) {
-            PerSamplePileupElementTracker<PileupElement> tracker = (PerSamplePileupElementTracker<PileupElement>) pileupElementTracker;
-            PerSamplePileupElementTracker<PileupElement> filteredTracker = new PerSamplePileupElementTracker<PileupElement>();
-
-            for (final String sample : tracker.getSamples()) {
-                PileupElementTracker<PileupElement> perSampleElements = tracker.getElements(sample);
-                ReadBackedPileupImpl pileup = createNewPileup(loc, perSampleElements).getOverlappingFragmentFilteredPileup(discardDiscordant, baseQualNotMapQual);
-                filteredTracker.addElements(sample, pileup.pileupElementTracker);
-            }
-            return createNewPileup(loc, filteredTracker);
-        } else {
-            Map<String, PileupElement> filteredPileup = new HashMap<String, PileupElement>();
+    public ReadPileup getOverlappingFragmentFilteredPileup(boolean discardDiscordant, boolean baseQualNotMapQual) {
```

Comment:
It might be nicer to have this take a function for getting the tiebreaker instead of the boolean.  I.e. 

public ReadPileup getOverlappingFragmentFilteredPileup(boolean discardDiscordant, Function<ReadPileup, Integer> tieBreaker)